app/mutations/addresses/release.rb

```diff
@@ -31,10 +32,18 @@ def execute
         info "Skip gateway address=#{address.id} in pool=#{@pool.id}"
 
       else
-        info "Delete address=#{address.id} in pool=#{@pool.id}"
-
-        address.delete!
+        if ping?(address.address)
+          info "#{self.address} still responds to ping, deleting omitted"
```

Comment:
Is it really a good idea to make this error out the API call on agent side? Isn't this kind-of deferred release so from agent point-of view it would still mean successful API call?